ee/app/livechat-enterprise/server/hooks/beforeRoutingChat.js

```diff
@@ -3,10 +3,29 @@ import { settings } from '../../../../../app/settings/server';
 import { LivechatInquiry } from '../../../../../app/models/server';
 import { dispatchInquiryPosition } from '../lib/Helper';
 import { allowAgentSkipQueue } from '../../../../../app/livechat/server/lib/Helper';
+import { Livechat } from '../../../../../app/livechat/server/lib/Livechat';
+import { LivechatDepartment, LivechatInquiry as LivechatInquiryRaw, LivechatRooms } from '../../../../../app/models/server/raw';
+import { online } from '../../../../../app/livechat/server/api/lib/livechat';
 import { saveQueueInquiry } from '../../../../../app/livechat/server/lib/QueueManager';
 import { cbLogger } from '../lib/logger';
 
 callbacks.add('livechat.beforeRouteChat', async (inquiry, agent) => {
+	// check here if department has fallback before queueing
+	if (inquiry?.department && !online(inquiry.department, true)) {
+		cbLogger.debug('No agents online on selected department. Inquiry will use fallback department');
+		const department = await LivechatDepartment.findOneById(inquiry.department);
+		if (department.fallbackForwardDepartment) {
+			cbLogger.debug(`Inquiry ${ inquiry._id } will be moved from department ${ department._id } to fallback department ${ department.fallbackForwardDepartment }`);
+			// update visitor
+			Livechat.setDepartmentForGuest({ token: inquiry?.v?.token, department: department.fallbackForwardDepartment });
```

Comment:
I noticed that if I start a chat with a department that is offline, and then when chat is transferred to the fallback department - the department is not visible on the Room UI
![image](https://user-images.githubusercontent.com/34130764/146772435-f1cc109c-4f1b-4e0a-bd7f-19f220f95ebb.png)

So I think we need to update the room department property here as well

Reply:
Pls try again with latest changes. It was wrongly updating the department property (cause in Room is departmentId) and that could have been causing the issue.

Reply:
Checked and its working now